### ISSUE TITLE
Rule proposal: `utf-8-encoding-case`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1690]
### Description

Enforce consistent style `"UTF-8"` encoding.

In many repos, `'utf8'` and `'utf-8'` both used. I'm not sure what's the best choice, I believe `UTF-8` is the standard name, but Node.js docs recommended `utf8` in many places. Maybe configurable?

### Fail

```js
fs.readFile(foo, 'utf-8');
```

```js
new TextDecoder('utf8');
```

### Pass

```js
fs.readFile(foo, 'UTF-8');
```

```js
new TextDecoder('UTF-8');
```

**Comment by sindresorhus** [comment]
I prefer `utf8`. This is what most Node.js users would use too. Yes, `UTF-8` is the official name, but in this context, it's used more like an identifier / enum case. JavaScript doesn't have enums, but we instead use string unions. The conventions for these is that they're formatted as if they were proper identifiers.

**Comment by sindresorhus** [comment]
I also think the rule name should be `utf8-encoding-case`.

**Comment by fisker** [comment]
I'm fine with `utf8`, main point is to avoid different cases.

**Comment by silverwind** [comment]
Does Node.js accept `UTF-8` for `encoding`?

**Comment by silverwind** [comment]
For reference, here are the spec details on the TextEncoder argument:

https://encoding.spec.whatwg.org/#concept-encoding-get

**Comment by silverwind** [comment]
I think the rule should be called `encoding-case` to support other encodings besides UTF-8 as well.

**Comment by sindresorhus** [comment]
I don't really see the point of complicating the rule with any possible encoding. The 99.99% case is UTF-8.

**Comment by silverwind** [comment]
There are still many cases where one has to deal with `ascii` encoding, especially when dealing with network protocols like DNS in Node.js. Those protocols will never get UTF-8 support for compatibility reasons. I think it's short-sighted to only support one encoding.

**Comment by papb** [comment]
So only `utf8` and `ascii`, then?

**Comment by sindresorhus** [comment]
I'm fine with including `ascii` too.

**Comment by fisker** [comment]
My plan is to report all string literals that looks like a `'utf8'` or `'ascii'` encoding, no matter where it's used.

**Comment by fisker** [comment]
`encoding-case`?

**Comment by sindresorhus** [comment]
I would go with `text-encoding-identifier-case`.

**Comment by sindresorhus** [comment]
Accepted

**Comment by fisker** [comment]
I found something in the WHATWG spec, I hope use `utf8` on the web is fine, https://encoding.spec.whatwg.org/#names-and-labels

> Authors must use the [UTF-8](https://encoding.spec.whatwg.org/#utf-8) [encoding](https://encoding.spec.whatwg.org/#encoding) and must use the [ASCII case-insensitive](https://infra.spec.whatwg.org/#ascii-case-insensitive) "utf-8" [label](https://encoding.spec.whatwg.org/#label) to identify it.

**Comment by domoritz** [comment]
I am getting false positives in my jest test definitions with this rule:

```
describe('Utf8', () => { validateVector(generate.utf8()); });
```

**Comment by pcorpet** [comment]
According to [WhatWG](https://html.spec.whatwg.org/multipage/semantics.html#character-encoding-declaration), in HTML spec the `<meta charset>` tag must be an ASCII case-insensitive match for the string "utf-8".

This is causing conflicts in our React code (in `.tsx` files that are linted by `eslint`):
```tsx
  return <html lang="en">
    <head>
      <link rel="icon" type="image/x-icon" href={favicon} />
      <meta charSet="utf-8" />
```

**Comment by Primajin** [comment]
Yeah I'm facing the same issue like @pcorpet - is there a way to make the rule aware to not apply this to JSX / meta charset?

**Comment by fisker** [comment]
I'll fix that.

**Comment by fisker** [comment]
https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1817

**Comment by Primajin** [comment]
> #1817

Thanks a bunch!
